classes/security/Validation.php

```diff
@@ -38,6 +37,33 @@ class Validation
     public const AUTH_KEY_USERNAME = 1;
     public const AUTH_KEY_EMAIL = 2;
 
+    protected static $authKey = self::AUTH_KEY_USERNAME;
+
+    /**
+     * Get the user currently set auth key
+     * 
+     * @return int const value of AUTH_KEY_* define auth key(email/username)
+     */
+    public static function getAuthKey(): int
+    {
+        return static::$authKey;
+    }
+
+    /**
+     * Set the user current auth key
+     * 
+     * @param int $value const value of AUTH_KEY_* define auth key(email/username)
+     * @return void
+     */
+    public static function setAuthKey(int $value): void
+    {
+        if (!in_array($value, [static::AUTH_KEY_USERNAME, static::AUTH_KEY_EMAIL])) {
```

Comment:
I am thinking about removing this part of code . Basically this keep track if user logged in via email or username and autofill the auth key field with appropriate information for next time login . But as we are going to promote to use email more for login purpose , it's ok when someone login via username and next time it autofill the auth key with email , what you say ?

Reply:
I think it's better to have a single source of truth, as the email can be changed, the username seems to be better.
ps: I didn't inspect much how these `AUTH_KEY_*` constants are being used, after skimming the code, looks like they can be discarded/replaced by code.

Reply:
@touhidurabir, is this resolved to your satisfaction?

Reply:
> is this resolved to your satisfaction

@asmecher truthfully no . what I like to propose/do here is to remove this whole part of having/determining `AUTH_KEY_*` part . What it do is when user login using the `username` and then logout, and then again login , the login form will have user email instead of username auto filled . As we are going to more towards email instead of username to login, only to have it show the last username/email (specific one) auto filled and to do that, have these much redundant code seems really unnecessary to me . what you say ? 

This will also remove the entanglement between the `Validation` and `PKPUserProvider` and we also aim to remove the `PKP\security\Validation` in future so bit advance clean up .

Reply:
I have removed those part of codes as they have no use with our current implementation.